### DISCUSSION TITLE
How pass id model to method feature test

### DISCUSSION THREAD

**Discussion by pixsolution** [livewire/livewire#5626] [1. Help]
Hi, how can I test a simply delete method like this one:

```php
public function delete(Form $form)
    {
        if ($form) {
            $form->delete();
            session()->flash('message', 'Formulario eliminado con éxito');
        }
    }
```

my test:

```php
/** @test */
    public function user_can_delete_form()
    {
        $user = User::factory()->create();

        $form = Form::factory()->create(['user_id' => $this->user->id]);

        $this->actingAs($user);

        $component = Livewire::test(Forms::class, ['form' => $form]);
            $component->call('delete');

        $this->assertModelMissing($form);

    }
```

But it failed because The data from my test does not reach the method, if I do dd($form) in the method I get null $form
![1](https://user-images.githubusercontent.com/47191848/227543083-0ffea2af-ea7b-4c85-bc6b-036c562e9b8e.jpg)

 and the method is working because by the browser if I do dd($form) if I get the object right. 
![2](https://user-images.githubusercontent.com/47191848/227543538-0df1f01e-ca02-40be-9da1-592bf87e4154.jpg)

Also validate that the factory was registering correctly and if it does, the record is output. So I think it's like passing the test parameter to the method. I've also tried Livewire::withQueryParams(['form' => $form]) and nothing, same result. What can be?

**Comment by joshhanley** [comment]
@pixsolution you can pass extra parameters to the `call()` method, like `->call('delete', $form)`.

Hope this helps!
